docs/source/tutorial/resolvers.md

```diff
@@ -3,10 +3,369 @@ title: "3. Write your graph's resolvers"
+
+Start by copying the query below and pasting it in the left side of the playground. After pasting, hit the Play button at the center to get a response.
+
+```js
+query {
+  launches {
+    id
+    year
+    mission {
+      name
+    }
+  }
+}
+```
+
+A response like this should be returned at the right side of the playground.
+
+![All Launches](../images/launches.png)
+
+Let's write a launch query that accepts an argument. Copy the query below and paste it in the playground. Hit the Play button to get a response.
+
+```js
+query {
+  launch(id: 60) {
+    id
+    year
+    rocket {
+      id
+      name
+      type
+    }
+  }
+}
+```
+
+A response like the one shown below should be returned at the right side of the playground.
+
+![A Launch](../images/alaunch.png)
+
+
+<h2 id="pagination">Pagination</h2>
+
+The `launches` query returned a large data set of launches. This data set contains too much data to be fetched all at once.
+
+Pagination is a solution to the challenge of fetching all the data at once. It's a technique that ensures that the server only sends data in small chunks per time. There are two major ways of fetching paginated data: offset-based (otherwise known as numbered pages), and cursor-based pagination.
+
+Cursor-based pagination is the more efficient way of fetching paginated data because it eliminates the possibility of skipping items and displaying the same item more than once. In cursor-based pagination, a constant pointer is used to keep track of where in the data set the next items should be fetched from. This pointer is called a **cursor**.
+
+We'll use a cursor-based pagination for our graph API. This approach requires two parameters:
+
+* The number of items to fetch at once.
+* The cursor.
+
+Open up the `src/schema.js` file and update the `Query` type. Go ahead and add a new type called `LaunchConnection` to the schema as shown below:
+
+_src/schema.js_
+
+```js
+type Query {
+  launches(
+    """
+    The number of results to show. Must be >= 1. Default = 20
+    """
+    pageSize: Int
+    """
+    If you add a cursor here, it will only return results _after_ this cursor
+    """
+    after: String
+  ): LaunchConnection!
+  launch(id: ID!): Launch
+  me: User
+}
+
+"""
+Simple wrapper around our list of launches that contains a cursor to the
+last item in the list. Pass this cursor to the launches query to fetch results
+after these.
+"""
+type LaunchConnection {
+  cursor: String!
+  hasMore: Boolean!
+  launches: [Launch]!
+}
+...
+```
+
+The `launches` field takes in two parameters, `pageSize` and `after`. `pageSize` refers to the number of items to show at once while `after` refers to the cursor that keeps track of where the next set of data should be fetched from. We created a `LaunchConnection` type that returns a result that shows the list of launches with a `cursor` field and a `hasMore` field to indicate if there's more data to be fetched.
```

Comment:
Can we add a quick sentence explaining docstrings? I don't think we've introduced them until this point. Also, can you please change the markdown syntax highlighting at the top of the code block to `graphql` instead of `js`?